File: app/src/main/java/com/kakacat/minitool/expressinquiry/activity/DeliveryDetailActivity.java
package com.kakacat.minitool.expressinquiry.activity;

import android.graphics.Color;
import android.os.Bundle;
import android.view.MenuItem;
import android.view.View;
import android.view.Window;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.appbar.CollapsingToolbarLayout;
import com.kakacat.minitool.R;
import com.kakacat.minitool.expressinquiry.adapter.DeliveryDetailAdapter;
import com.kakacat.minitool.expressinquiry.model.Delivery;

public class DeliveryDetailActivity extends AppCompatActivity {

    private Delivery mDelivery;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_delivery_detail);

        initView();
    }

    private void initView() {
        Window window = getWindow();
        window.getDecorView().setSystemUiVisibility(View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN | View.SYSTEM_UI_FLAG_LAYOUT_STABLE);
        window.setStatusBarColor(Color.TRANSPARENT);

        mDelivery = (Delivery) getIntent().getSerializableExtra("delivery");
        initToolbar();

        RecyclerView recyclerView = findViewById(R.id.rv);

        assert mDelivery != null;
        DeliveryDetailAdapter adapter = new DeliveryDetailAdapter(mDelivery.getPackageDetailList());
        recyclerView.setAdapter(adapter);
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
    }

    private void initToolbar() {
        setSupportActionBar(findViewById(R.id.toolbar));
        CollapsingToolbarLayout toolbarLayout = findViewById(R.id.toolbar_layout);
        toolbarLayout.setTitle(mDelivery.getCompanyName());
        toolbarLayout.setCollapsedTitleTextColor(Color.WHITE);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setHomeAsUpIndicator(R.drawable.ic_action_back);
            actionBar.setDisplayShowTitleEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem menuItem) {
        if (menuItem.getItemId() == android.R.id.home) {
            finish();
        }
        return true;
    }
}

File: app/src/main/java/com/kakacat/minitool/expressinquiry/activity/DeliveryDetailActivity.kt
package com.kakacat.minitool.expressinquiry.activity

import android.graphics.Color
import android.os.Bundle
import android.view.View
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.appbar.CollapsingToolbarLayout
import com.kakacat.minitool.R
import com.kakacat.minitool.common.base.BaseActivity
import com.kakacat.minitool.common.util.UiUtil
import com.kakacat.minitool.expressinquiry.adapter.DeliveryDetailAdapter
import com.kakacat.minitool.expressinquiry.model.Delivery

class DeliveryDetailActivity : BaseActivity() {

    private var mDelivery: Delivery? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_delivery_detail)
        initView()
    }

    private fun initView() {
        UiUtil.setTranslucentStatusBarBlack(this)
        mDelivery = intent.getSerializableExtra("delivery") as Delivery
        initToolbar()
        val recyclerView = findViewById<RecyclerView>(R.id.rv)
        val adapter = DeliveryDetailAdapter(mDelivery!!.packageDetailList)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)
    }

    private fun initToolbar() {
        setSupportActionBar(findViewById(R.id.toolbar))
        val toolbarLayout = findViewById<CollapsingToolbarLayout>(R.id.toolbar_layout)
        toolbarLayout.title = mDelivery!!.companyName
        toolbarLayout.setCollapsedTitleTextColor(Color.WHITE)
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setHomeAsUpIndicator(R.drawable.ic_action_back)
            actionBar.setDisplayShowTitleEnabled(true)
            actionBar.setDisplayHomeAsUpEnabled(true)
        }
    }
}